app/helpers/templates_helper.rb

```diff
@@ -28,6 +28,20 @@ def infobox_functions_and_macros(docs_section)
           :rel => 'external')
   end
 
+  def safemode_methods
+    Rails.cache.fetch("safemode_methods", :expires_in => 2.hours) do
```

Comment:
The version number (`SETTING[:version]`) should probably be added to the cache key, then it'll automatically invalidate on update.